js/ui/codeView.js

```diff
@@ -453,6 +457,63 @@ var CodingSession = new Lang.Class({
         return this._toolbox;
     },
 
+    setGrabbed: function(grabbed) {
+        if (this._grabTimeoutId > 0) {
+            Mainloop.source_remove(this._grabTimeoutId);
+            this._grabTimeoutId = 0;
+        }
+
+        if (!this._button)
+            return;
+
+        if (grabbed) {
+            this._button.opacity = 0;
+        } else {
+            // giving some time to finish possible window movement FXs
+            this._grabTimeoutId = Mainloop.timeout_add(300,
+                () => {
+                    this._button.opacity = 255;
+                    return GLib.SOURCE_REMOVE;
+                });
+        }
+    },
+
+    _attachAppToToolbox: function() {
```

Comment:
This is a quite neat trick! However, from what I can see your change always sets up the app window actor to be a child of the toolbox while the toolbox is displayed.
I am a bit concerned that will break some expectation elsewhere that window actors are children of the window group.

On the other hand, if this does work reliably I think that we could remove a lot of the window position tracking/synchronization code that we have elsewhere in `codeView.js`, but that's probably besides the point of this PR.

Perhaps for now we could set up the nested hierarchy only while the window is being dragged?

Reply:
The first version of the path does that, but I changed to do it after the flip. I've implemented the original version again so now the attach is only done during the grab.